src/cmap/connection.ts

```diff
@@ -1145,12 +985,10 @@ export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
         clusterTime = session.clusterTime;
       }
 
-      const err = applySession(session, cmd, options);
-      if (err) {
-        return callback(err);
-      }
+      const sessionError = applySession(session, cmd, options);
```

Comment:
`applySession` is only called here - could we just make `applySession` throw instead?

Reply:
🤦 Duh.  ofc we still need it in the regular connection class